addons/project_todo/static/src/views/todo_form/todo_form_controller.js

```diff
@@ -0,0 +1,57 @@
+/** @odoo-module **/
+
+import { FormController } from "@web/views/form/form_controller";
+import { TodoEditableBreadcrumbName } from "@project_todo/components/todo_editable_breadcrumb_name/todo_editable_breadcrumb_name";
+import { TodoDoneCheckmark } from "@project_todo/components/todo_done_checkmark/todo_done_checkmark";
+
+import { onWillStart } from "@odoo/owl";
+
+/**
+ *  The FormController is overridden to be able to manage the edition of the name of a to-do directly
+ *  in the breadcrumb as well as the mark as done button next to it.
+ */
+
+export class TodoFormController extends FormController {
+    setup() {
+        super.setup();
+        onWillStart(async () => {
+            this.projectAccess = await this.model.orm.call(
+                'project.task',
+                'get_project_access_right',
+                [],
+            );
+        });
+    }
+    static template = "project_todo.TodoFormView";
+
+    get actionMenuItems() {
+        const actionToKeep = ["archive", "unarchive", "duplicate", "delete"];
+        const menuItems = super.actionMenuItems;
+        const filteredActions =
+            menuItems.action?.filter((action) => actionToKeep.includes(action.key)) || [];
+
+        if (this.projectAccess) {
+            filteredActions.push({
+                description: this.env._t("Convert to Task"),
+                callback: async () => {
```

Comment:
I think `async` is no longer required since we no longer do a rpc inside that callback function. 🙂